test/Wrapper/Wrapper.js

```diff
@@ -0,0 +1,146 @@
+const { expect } = require("chai");
+const { BigNumber } = require("ethers");
+const { ethers } = require("hardhat");
+
+let accounts;
+let ownerAddress;
+let tokensContract;
+let helperRoleContract;
+let wrapperContract;
+let erc20Contract;
+let spaceETH;
+
+describe("SpacePiratesWrapper", () => {
+  before(async () => {
+    const HelperRoleContract = await ethers.getContractFactory(
+      "HelperRoleContract"
+    );
+    helperRoleContract = await HelperRoleContract.deploy();
+
+    const TokenContract = await ethers.getContractFactory("SpacePiratesTokens");
+    tokensContract = await TokenContract.deploy("uri://");
+
+    accounts = await ethers.getSigners();
+    ownerAddress = accounts[0].getAddress();
+
+    const WrapperContract = await ethers.getContractFactory(
+      "SpacePiratesWrapper"
+    );
+    wrapperContract = await WrapperContract.deploy(tokensContract.address);
+
+    const ERC20Contract = await ethers.getContractFactory("TestToken");
+    erc20Contract = await ERC20Contract.deploy(10000);
+
+    spaceETH = await wrapperContract.spaceETH();
+
+    await tokensContract.setApprovalForAll(wrapperContract.address, true);
+  });
+  it("addERC20", async () => {
+    const lastId = await wrapperContract.lastId();
+
+    await wrapperContract.addERC20(erc20Contract.address);
+
+    expect(await wrapperContract.erc20ToId(erc20Contract.address)).to.equal(
+      BigNumber.from(lastId).add(1)
+    );
+  });
+  it("erc20Deposit", async () => {
+    const id = await wrapperContract.erc20ToId(erc20Contract.address);
+    const prevBalance = await tokensContract.balanceOf(ownerAddress, id);
+
+    const mintRole = await helperRoleContract.getMintRoleBytes(id);
+    await tokensContract.grantRole(mintRole, wrapperContract.address);
+
+    const amount = 10;
+
+    await erc20Contract.approve(wrapperContract.address, amount);
+    await wrapperContract.erc20Deposit(erc20Contract.address, amount);
+
+    expect(await tokensContract.balanceOf(ownerAddress, id)).to.equal(
+      BigNumber.from(prevBalance).add(amount)
+    );
```

Comment:
add check to the change of the balance before and after also of the erc20 token on wrapping contract and user